src/hardware/pros_motor.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include "constant/constants.h"
+#include "interface/encoder.h"
+#include "interface/motor.h"
+#include "pros.h"
+
+namespace hardware {
+
+enum class ProsMotorCartridge {
+  // equal to pros value to enable casting
+  kBlueCartridge = pros::E_MOTOR_GEAR_BLUE,   // 600 rpm
+  kGreenCartridge = pros::E_MOTOR_GEAR_GREEN  // 200 rpm
+};
+
+pros::motor_gearset_e_t ConvertProsMotorCartridge(ProsMotorCartridge);
+ProsMotorCartridge ConvertProsGearset(pros::motor_gearset_e_t);
+
+/**
+ * A helper function for getting the max rpm of a motor.
+ */
+int GetMaxRpm(const ProsMotorCartridge);
+
+/**
+ * A class which wraps a single pros::Motor object.
+ */
+class ProsMotor : public interface::Motor, public interface::Encoder {
+ public:
+  ProsMotor(int port_number, bool reverse, ProsMotorCartridge cartridge)
+      : motor_(
+            port_number * (reverse ? -1 : 1), ConvertProsMotorCartridge(cartridge)) {
```

Comment:
wait you can just reverse a motor by giving it a negative port number?

Reply:
Yeah, pros is weird.